File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.java
// Last updated: 5/29/2026, 11:57:19 AM
class Solution {
    public int  minSubArrayLen(int target,int[] nums){
        int left=0,right=0,ws=0,ms=Integer.MAX_VALUE;
        for(right=0;right<nums.length;right++){
            ws+=nums[right];
            while(ws>=target){
                ms=Math.min(ms,right-left+1);
                ws-=nums[left];
                left++;
            }
        }
        if(ms==Integer.MAX_VALUE){
            return 0;
        }
        return ms;
    }
    
}

File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.py
# Last updated: 5/29/2026, 11:57:13 AM
class Solution(object):
    def minSubArrayLen(self,target, nums):
        left = 0
        total = 0
        min_len = float('inf')

        for right in range(len(nums)):
            total += nums[right]
            while total >= target:
                min_len = min(min_len, right - left + 1)
                total -= nums[left]
                left += 1

        return 0 if min_len == float('inf') else min_len

        